spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+require 'shoulda-matchers'
+
+include Testable
+
+describe SessionsController do
+  it { should use_before_action :require_logout }
+
+  describe 'POST create' do
+    let(:user) { Fabricate :user }
+
+    context 'with valid credentials' do
+      before { post :create, email: user.email, password: user.password }
+
+      it 'sets the session user_id' do
+        expect(session[:user_id]).to eq user.id
+      end
+
+      it 'redirects to the home page on successfull login' do
+        expect(response).to redirect_to home_path
+      end
+
+      it 'sets flash[:notice]' do
+        expect(flash[:notice]).to be_a String
+      end
+    end
+
+    context 'with invalid credentials' do
+      before { post :create, email: user.email, password: user.password + '1' }
+
+      it 'does not set the session user_id' do
+        expect(session[:user_id]).to be nil
+      end
+
+      it 'renders the new template if user email is invalid' do
```

Comment:
"user email is invalid" is actually incorrect anyway. The password is incorrect.